### ISSUE TITLE
4 tests fail

### ISSUE THREAD

**Issue by yurivict** [rsheftel/pandas_market_calendars#275]
```
========================================================================================== FAILURES ===========================================================================================
____________________________________________________________________________________ test_good_friday_rule ____________________________________________________________________________________

    def test_good_friday_rule():
        # Good friday is a holiday unless Christmas Day or New Years Day is on a Friday
        for calendar in calendars:
    
            cal = calendar()
            valid_days = cal.valid_days('2015-04-01', '2016-04-01')
            for day in ["2015-04-03", "2016-03-25"]:
>               assert day in valid_days
E               AssertionError: assert '2015-04-03' in DatetimeIndex(['2015-04-01 00:00:00+00:00', '2015-04-02 00:00:00+00:00',\n               '2015-04-06 00:00:00+00:00', '2015-04-07 00:00:00+00:00',\n               '2015-04-08 00:00:00+00:00', '2015-04-09 00:00:00+00:00',\n               '2015-04-10 00:00:00+00:00', '2015-04-13 00:00:00+00:00',\n               '2015-04-14 00:00:00+00:00', '2015-04-15 00:00:00+00:00',\n               ...\n               '2016-03-18 00:00:00+00:00', '2016-03-21 00:00:00+00:00',\n               '2016-03-22 00:00:00+00:00', '2016-03-23 00:00:00+00:00',\n               '2016-03-24 00:00:00+00:00', '2016-03-28 00:00:00+00:00',\n               '2016-03-29 00:00:00+00:00', '2016-03-30 00:00:00+00:00',\n               '2016-03-31 00:00:00+00:00', '2016-04-01 00:00:00+00:00'],\n              dtype='datetime64[ns, UTC]', length=253, freq='C')

tests/test_cboe_calendars.py:50: AssertionError
________________________________________________________________________________ test_schedule_w_interruptions ________________________________________________________________________________

self = <tests.test_market_calendar.FakeCalendar object at 0x9162a67c0>
col = 2002-02-03        None
2010-01-11        None
2010-01-13    10:29:00
2011-01-10        None
Name: interruption_start_2, dtype: object

    def _convert(self, col):
>       try: times = col.str[0]

pandas_market_calendars/market_calendar.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/pandas/core/generic.py:5989: in __getattr__
    return object.__getattribute__(self, name)
/usr/local/lib/python3.9/site-packages/pandas/core/accessor.py:224: in __get__
    accessor_obj = self._accessor(obj)
/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:181: in __init__
    self._inferred_dtype = self._validate(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = array([None, None, datetime.time(10, 29), None], dtype=object)

    @staticmethod
    def _validate(data):
        """
        Auxiliary function for StringMethods, infers and checks dtype of data.
    
        This is a "first line of defence" at the creation of the StringMethods-
        object, and just checks that the dtype is in the
        *union* of the allowed types over all string methods below; this
        restriction is then refined on a per-method basis using the decorator
        @forbid_nonstring_types (more info in the corresponding docstring).
    
        This really should exclude all series/index with any non-string values,
        but that isn't practical for performance reasons until we have a str
        dtype (GH 9343 / 13877)
    
        Parameters
        ----------
        data : The content of the Series
    
        Returns
        -------
        dtype : inferred dtype of data
        """
        if isinstance(data, ABCMultiIndex):
            raise AttributeError(
                "Can only use .str accessor with Index, not MultiIndex"
            )
    
        # see _libs/lib.pyx for list of inferred types
        allowed_types = ["string", "empty", "bytes", "mixed", "mixed-integer"]
    
        data = extract_array(data)
    
        values = getattr(data, "categories", data)  # categorical / normal
    
        inferred_dtype = lib.infer_dtype(values, skipna=True)
    
        if inferred_dtype not in allowed_types:
>           raise AttributeError("Can only use .str accessor with string values!")
E           AttributeError: Can only use .str accessor with string values!

/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:235: AttributeError

During handling of the above exception, another exception occurred:

>   ???

pandas/_libs/tslibs/timedeltas.pyx:440: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Expected unicode, got NAType

pandas/_libs/tslibs/timedeltas.pyx:472: TypeError

During handling of the above exception, another exception occurred:

    def test_schedule_w_interruptions():
    
        cal = FakeCalendar()
    
>       results = cal.schedule("2010-01-08", "2010-01-14", interruptions= True, tz= cal.tz)

tests/test_market_calendar.py:751: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas_market_calendars/market_calendar.py:637: in schedule
    interrs = self.interruptions_df
pandas_market_calendars/market_calendar.py:452: in interruptions_df
    return intr.apply(self._convert).sort_index()
/usr/local/lib/python3.9/site-packages/pandas/core/frame.py:9423: in apply
    return op.apply().__finalize__(self, method="apply")
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:678: in apply
    return self.apply_standard()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:798: in apply_standard
    results, res_index = self.apply_series_generator()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:814: in apply_series_generator
    results[i] = self.f(v)
pandas_market_calendars/market_calendar.py:427: in _convert
    return (pd.to_timedelta(col.astype("string"), errors="coerce") + col.index
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:189: in to_timedelta
    values = _convert_listlike(arg._values, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:248: in _convert_listlike
    td64arr = sequence_to_td64ns(arg, unit=unit, errors=errors, copy=False)[0]
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:912: in sequence_to_td64ns
    data = _objects_to_td64ns(data, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:1041: in _objects_to_td64ns
    result = array_to_timedelta64(values, unit=unit, errors=errors)
pandas/_libs/tslibs/timedeltas.pyx:454: in pandas._libs.tslibs.timedeltas.array_to_timedelta64
    ???
pandas/_libs/tslibs/timedeltas.pyx:484: in pandas._libs.tslibs.timedeltas._item_to_timedelta64
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Invalid type for timedelta scalar: <class 'pandas._libs.missing.NAType'>

pandas/_libs/tslibs/timedeltas.pyx:371: TypeError
_______________________________________________________________________________ test_open_at_time_interruptions _______________________________________________________________________________

self = <tests.test_market_calendar.FakeBreakCalendar object at 0x92c795850>
col = 2010-01-11        None
2010-01-12    11:55:00
2010-01-13    11:05:00
Name: interruption_start_2, dtype: object

    def _convert(self, col):
>       try: times = col.str[0]

pandas_market_calendars/market_calendar.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/pandas/core/generic.py:5989: in __getattr__
    return object.__getattribute__(self, name)
/usr/local/lib/python3.9/site-packages/pandas/core/accessor.py:224: in __get__
    accessor_obj = self._accessor(obj)
/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:181: in __init__
    self._inferred_dtype = self._validate(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = array([None, datetime.time(11, 55), datetime.time(11, 5)], dtype=object)

    @staticmethod
    def _validate(data):
        """
        Auxiliary function for StringMethods, infers and checks dtype of data.
    
        This is a "first line of defence" at the creation of the StringMethods-
        object, and just checks that the dtype is in the
        *union* of the allowed types over all string methods below; this
        restriction is then refined on a per-method basis using the decorator
        @forbid_nonstring_types (more info in the corresponding docstring).
    
        This really should exclude all series/index with any non-string values,
        but that isn't practical for performance reasons until we have a str
        dtype (GH 9343 / 13877)
    
        Parameters
        ----------
        data : The content of the Series
    
        Returns
        -------
        dtype : inferred dtype of data
        """
        if isinstance(data, ABCMultiIndex):
            raise AttributeError(
                "Can only use .str accessor with Index, not MultiIndex"
            )
    
        # see _libs/lib.pyx for list of inferred types
        allowed_types = ["string", "empty", "bytes", "mixed", "mixed-integer"]
    
        data = extract_array(data)
    
        values = getattr(data, "categories", data)  # categorical / normal
    
        inferred_dtype = lib.infer_dtype(values, skipna=True)
    
        if inferred_dtype not in allowed_types:
>           raise AttributeError("Can only use .str accessor with string values!")
E           AttributeError: Can only use .str accessor with string values!

/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:235: AttributeError

During handling of the above exception, another exception occurred:

>   ???

pandas/_libs/tslibs/timedeltas.pyx:440: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Expected unicode, got NAType

pandas/_libs/tslibs/timedeltas.pyx:472: TypeError

During handling of the above exception, another exception occurred:

    def test_open_at_time_interruptions():
        cal = FakeBreakCalendar()
>       sched = cal.schedule("2010-01-08", "2010-01-14", interruptions=True)

tests/test_market_calendar.py:1016: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas_market_calendars/market_calendar.py:637: in schedule
    interrs = self.interruptions_df
pandas_market_calendars/market_calendar.py:452: in interruptions_df
    return intr.apply(self._convert).sort_index()
/usr/local/lib/python3.9/site-packages/pandas/core/frame.py:9423: in apply
    return op.apply().__finalize__(self, method="apply")
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:678: in apply
    return self.apply_standard()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:798: in apply_standard
    results, res_index = self.apply_series_generator()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:814: in apply_series_generator
    results[i] = self.f(v)
pandas_market_calendars/market_calendar.py:427: in _convert
    return (pd.to_timedelta(col.astype("string"), errors="coerce") + col.index
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:189: in to_timedelta
    values = _convert_listlike(arg._values, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:248: in _convert_listlike
    td64arr = sequence_to_td64ns(arg, unit=unit, errors=errors, copy=False)[0]
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:912: in sequence_to_td64ns
    data = _objects_to_td64ns(data, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:1041: in _objects_to_td64ns
    result = array_to_timedelta64(values, unit=unit, errors=errors)
pandas/_libs/tslibs/timedeltas.pyx:454: in pandas._libs.tslibs.timedeltas.array_to_timedelta64
    ???
pandas/_libs/tslibs/timedeltas.pyx:484: in pandas._libs.tslibs.timedeltas._item_to_timedelta64
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Invalid type for timedelta scalar: <class 'pandas._libs.missing.NAType'>

pandas/_libs/tslibs/timedeltas.pyx:371: TypeError
____________________________________________________________________________________ test_interruptions_df ____________________________________________________________________________________

self = <tests.test_market_calendar.FakeCalendar object at 0x927fa99d0>
col = 2002-02-03        None
2010-01-11        None
2010-01-13    10:29:00
2011-01-10        None
Name: interruption_start_2, dtype: object

    def _convert(self, col):
>       try: times = col.str[0]

pandas_market_calendars/market_calendar.py:425: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.9/site-packages/pandas/core/generic.py:5989: in __getattr__
    return object.__getattribute__(self, name)
/usr/local/lib/python3.9/site-packages/pandas/core/accessor.py:224: in __get__
    accessor_obj = self._accessor(obj)
/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:181: in __init__
    self._inferred_dtype = self._validate(data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = array([None, None, datetime.time(10, 29), None], dtype=object)

    @staticmethod
    def _validate(data):
        """
        Auxiliary function for StringMethods, infers and checks dtype of data.
    
        This is a "first line of defence" at the creation of the StringMethods-
        object, and just checks that the dtype is in the
        *union* of the allowed types over all string methods below; this
        restriction is then refined on a per-method basis using the decorator
        @forbid_nonstring_types (more info in the corresponding docstring).
    
        This really should exclude all series/index with any non-string values,
        but that isn't practical for performance reasons until we have a str
        dtype (GH 9343 / 13877)
    
        Parameters
        ----------
        data : The content of the Series
    
        Returns
        -------
        dtype : inferred dtype of data
        """
        if isinstance(data, ABCMultiIndex):
            raise AttributeError(
                "Can only use .str accessor with Index, not MultiIndex"
            )
    
        # see _libs/lib.pyx for list of inferred types
        allowed_types = ["string", "empty", "bytes", "mixed", "mixed-integer"]
    
        data = extract_array(data)
    
        values = getattr(data, "categories", data)  # categorical / normal
    
        inferred_dtype = lib.infer_dtype(values, skipna=True)
    
        if inferred_dtype not in allowed_types:
>           raise AttributeError("Can only use .str accessor with string values!")
E           AttributeError: Can only use .str accessor with string values!

/usr/local/lib/python3.9/site-packages/pandas/core/strings/accessor.py:235: AttributeError

During handling of the above exception, another exception occurred:

>   ???

pandas/_libs/tslibs/timedeltas.pyx:440: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Expected unicode, got NAType

pandas/_libs/tslibs/timedeltas.pyx:472: TypeError

During handling of the above exception, another exception occurred:

    def test_interruptions_df():
    
        goal = pd.DataFrame({
                'interruption_start_1':
                    pd.Series(['2002-02-02 03:00:00', '2010-01-11 03:00:00',
                               '2010-01-13 01:59:00', '2011-01-10 03:00:00'],dtype= 'datetime64[ns]').dt.tz_localize('UTC'),
                'interruption_end_1':
                    pd.Series(['2002-02-03 03:02:00', '2010-01-12 03:01:00',
                               '2010-01-13 02:00:00', '2011-01-10 03:01:00'],dtype= 'datetime64[ns]').dt.tz_localize('UTC'),
                'interruption_start_2':
                    pd.Series([np.nan, np.nan, '2010-01-13 02:29:00', np.nan],
                              dtype= 'datetime64[ns]').dt.tz_localize('UTC'),
                'interruption_end_2':
                    pd.Series([np.nan, np.nan, '2010-01-13 02:30:00', np.nan],
                              dtype= 'datetime64[ns]').dt.tz_localize('UTC')
            }).set_index(pd.DatetimeIndex(['2002-02-03', '2010-01-11', '2010-01-13', '2011-01-10']))
    
        cal = FakeCalendar()
>       assert_frame_equal(cal.interruptions_df, goal)

tests/test_market_calendar.py:1101: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas_market_calendars/market_calendar.py:452: in interruptions_df
    return intr.apply(self._convert).sort_index()
/usr/local/lib/python3.9/site-packages/pandas/core/frame.py:9423: in apply
    return op.apply().__finalize__(self, method="apply")
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:678: in apply
    return self.apply_standard()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:798: in apply_standard
    results, res_index = self.apply_series_generator()
/usr/local/lib/python3.9/site-packages/pandas/core/apply.py:814: in apply_series_generator
    results[i] = self.f(v)
pandas_market_calendars/market_calendar.py:427: in _convert
    return (pd.to_timedelta(col.astype("string"), errors="coerce") + col.index
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:189: in to_timedelta
    values = _convert_listlike(arg._values, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/tools/timedeltas.py:248: in _convert_listlike
    td64arr = sequence_to_td64ns(arg, unit=unit, errors=errors, copy=False)[0]
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:912: in sequence_to_td64ns
    data = _objects_to_td64ns(data, unit=unit, errors=errors)
/usr/local/lib/python3.9/site-packages/pandas/core/arrays/timedeltas.py:1041: in _objects_to_td64ns
    result = array_to_timedelta64(values, unit=unit, errors=errors)
pandas/_libs/tslibs/timedeltas.pyx:454: in pandas._libs.tslibs.timedeltas.array_to_timedelta64
    ???
pandas/_libs/tslibs/timedeltas.pyx:484: in pandas._libs.tslibs.timedeltas._item_to_timedelta64
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: Invalid type for timedelta scalar: <class 'pandas._libs.missing.NAType'>

pandas/_libs/tslibs/timedeltas.pyx:371: TypeError
========================================================================= 4 failed, 1232 passed in 937.06s (0:15:37) ==========================================================================
```

Version: 4.2.1
Python-3.9
FreeBSD 13.2

**Comment by rsheftel** [comment]
Yes those are failing and we are looking into it

**Comment by rsheftel** [comment]
Fixed in PR #279
